pac4j-kerberos/src/main/java/org/pac4j/kerberos/credentials/authenticator/KerberosAuthenticator.java

```diff
@@ -0,0 +1,61 @@
+package org.pac4j.kerberos.credentials.authenticator;
+
+import java.util.HashMap;
+
+import org.pac4j.kerberos.credentials.KerberosCredentials;
+import org.pac4j.kerberos.profile.KerberosProfile;
+import org.pac4j.core.profile.CommonProfile;
+import org.pac4j.core.profile.ProfileHelper;
+import org.pac4j.core.profile.UserProfile;
+import org.pac4j.core.profile.creator.AuthenticatorProfileCreator;
+import org.pac4j.core.credentials.authenticator.Authenticator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Authenticator for Kerberos. It creates the user profile and stores it in the credentials
+ * for the {@link AuthenticatorProfileCreator}.
+ *
+ * @author Garry Boyce
+ * @since 1.9.1
+ */
+public class KerberosAuthenticator implements Authenticator<KerberosCredentials> {
+
+    protected final Logger                logger = LoggerFactory.getLogger(getClass());
+
+    private final KerberosTicketValidator ticketValidator;
+
+    /**
+     * Initializes the authenticator that will validate Kerberos tickets.
+     *
+     * @param ticketValidator    The ticket validator used to validate the Kerberos ticket.
+     * @since 1.9.1
+     */
+    public KerberosAuthenticator(KerberosTicketValidator ticketValidator) {

```

Comment:
For Shiro, we need an empty constructor and a setter/getter for the `ticketValidator`.

You need to inherit from `InitializableWebObject` as well and implement a `internalInit` method to check that the `ticketValidator` is not null.